lib/monkey/include/monkey/mk_core/mk_list.h

```diff
@@ -76,10 +76,10 @@ static inline void mk_list_add_after(struct mk_list *_new,
     }
 
     next = prev->next;
-    next->prev = prev;
     _new->next = next;
     _new->prev = prev;
     prev->next = _new;
+    next->prev = _new; /* how is this a bug not found? is it really? */
```

Comment:
Now resolved in monkey: https://github.com/monkey/monkey/commits?author=matthewfala